crates/oxc_linter/src/rules/typescript/prefer_enum_initializers.rs

```diff
@@ -0,0 +1,104 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("typescript-eslint(prefer-enum-initializers):")]
+#[diagnostic(severity(warning), help(""))]
+struct PreferEnumInitializersDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferEnumInitializers;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Require each enum member value to be explicitly initialized.
+    ///
+    /// ### Why is this bad?
+    /// In projects where the value of `enum` members are important, allowing implicit values for enums can cause bugs if enums are modified over time.
+    ///
+    /// ### Example
+    /// ```typescript
+    /// // wrong, the value of `Close` is not constant
+    /// enum Status {
+    ///  Open = 1,
+    ///  Close,
+    /// }
+    /// ```
+    PreferEnumInitializers,
+    correctness
```

Comment:
I prefer [`pedantic`](https://github.com/oxc-project/oxc/blob/0185eb2edcb3b01afb701cdab34643a26c2ad17e/crates/oxc_cli/src/command/lint.rs#L82-L83). As the document says, some care about enums having implicit values. 🤔

https://typescript-eslint.io/rules/prefer-enum-initializers/#when-not-to-use-it

Reply:
i've used `pedantic`, although I do not necessarily agree. this rule will very much help avoid errors with enum members being move around.